week2/index.js

```diff
@@ -0,0 +1,119 @@
+Added Abdullah
+
+"use strict";
+
+const fs = require('fs');
+const TODOS_FILENAME = process.env["TODOS_FILENAME"] || "todos.json";
+const helpFile = "help.txt";
+
+let commandArgs = process.argv.slice(2);
+let myTodoList = [];
+let num = 1;
+
+
+fs.readFile(TODOS_FILENAME, "utf8", (err, data) => {
+
+    if (err) {
+        //console.log(err);
+        if (err.code === "ENDENT") {
+
+            //No such File or directory
+            console.log("i'm sorry, friend, but that file doesn't exist.");
+        }
+        else if (err.code === "EACCES") {
+            console.log("That file exists but you don't have permission.");
+
+        }
+        else {
+
+            console.log(`Uh oh, an unknown error occured ${err.message}`)
+        }
+        process.exit();
```

Comment:
You should reject the promise here (instead of exiting) and let the caller decide how to handle the error through the `.catch()` block.

Reply:
i thought that we dont need to use promises to solve the homework that's why i didn't use it at all, but i will try to do it again using promises